src/NServiceBus.TransactionalSession/TransactionalSession.cs

```diff
@@ -1,33 +1,79 @@
 ﻿namespace NServiceBus.TransactionalSession
 {
-    using System.Collections.Generic;
     using System.Threading;
     using System.Threading.Tasks;
+    using Features;
+    using Microsoft.Extensions.DependencyInjection;
+    using Outbox;
     using Persistence;
     using Transport;
 
-    class TransactionalSession : TransactionalSessionBase
+    /// <summary>
+    /// Provides <see cref="ITransactionalSession" />.
+    /// </summary>
+    public class TransactionalSession : Feature
     {
-        public TransactionalSession(
-            ICompletableSynchronizedStorageSession synchronizedStorageSession,
-            IMessageSession messageSession,
-            IMessageDispatcher dispatcher,
-            IEnumerable<IOpenSessionOptionsCustomization> customizations) : base(synchronizedStorageSession, messageSession, dispatcher, customizations)
+        /// <summary>
+        /// See <see cref="Feature.Setup" />.
+        /// </summary>
+        protected override void Setup(FeatureConfigurationContext context)
         {
-        }
+            QueueAddress localQueueAddress = context.LocalQueueAddress();
 
-        protected override async Task CommitInternal(CancellationToken cancellationToken = default)
-        {
-            await synchronizedStorageSession.CompleteAsync(cancellationToken).ConfigureAwait(false);
+            var isOutboxEnabled = context.Settings.IsFeatureActive(typeof(Outbox));
```

Comment:
would this require an `DependsOnOptionally` (or however this is called) to ensure ordering?

Reply:
Done